src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,34 +1,100 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Prop = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Prop> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [movie, setMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const conditionToReject =
+    !movie.title.trim() ||
+    !movie.imgUrl.trim() ||
+    !movie.imdbUrl.trim() ||
+    !movie.imdbId.trim();
+
+  const handleChange = (name: string, newValue: string) => {
+    setMovie(newMovie => ({
+      ...newMovie,
+      [name]: newValue,
+    }));
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (conditionToReject) {
+      return;
+    } else {
+      onAdd({
+        ...movie,
+      });
+
+      setCount(count + 1);
+    }
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      action="src/api/movies.json"
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={movie.title}
+        onChange={newValue => handleChange('title', newValue)}
```

Comment:
```suggestion
        onChange={handleChange}
```
Consider reusing just one function that accepts the event object instead of creating 5 different functions